lib/virt_autotest/utils.pm

```diff
@@ -105,10 +105,23 @@ sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
+#check kernels by grep keywords from journals
+#welcome everybody to extend this function
+sub check_kernel {
+    my $machine //= 'localhost';
+
+    my @oops_keywords = ('Started Process Core Dump', 'Call Trace');
+    foreach my $keyword (@oops_keywords) {
+        my $cmd = "journalctl | grep '$keyword'";
+        $cmd = "ssh root\@$machine " . "\"$cmd\"" if $machine ne 'localhost';
+        record_soft_failure("$keyword in $machine fg. bsc#1204727") if script_run($cmd) == 0;
+    }
+}
+
 #check host to make sure it works well
 #welcome everybody to extend this function
 sub check_host {
-
+    check_kernel;
```

Comment:
It looks to me that what you do in check_kernel does not fulfill your purpose completely. After confirming there is keyword, why not dig into it further ? For example, detect crash or print out something ?

Reply:
The logging system will upload the coredumps and journals finally. Meanwhile, it is hard to tell a machine crashed.